src/Cli/dotnet/commands/dotnet-test/TestingPlatformCommand.cs

```diff
@@ -54,7 +87,9 @@ public int Run(ParseResult parseResult)
                     testApp.Run += OnTestApplicationRun;
                     testApp.ExecutionIdReceived += OnExecutionIdReceived;
 
-                    return await testApp.RunAsync(filterModeEnabled, enableHelp: true, builtInOptions);
+                    var result = await testApp.RunAsync(filterModeEnabled, enableHelp: true, builtInOptions);
+                    CompleteRun();
```

Comment:
CompleteRun() is only added for the help option, it should be added for the else block as well.